### ISSUE TITLE
Incorrect type evaluation for tuple constructor call

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7116]
The following should not generate an error, but it does.

```python
def func4(untyped_iter: Iterable[Any]):
    a: tuple[int, int] = tuple(untyped_iter)  # OK
```

**Comment by erictraut** [comment]
Duplicate of #7085.
